### ISSUE TITLE
Allow to group actions

### ISSUE THREAD

**Issue by BorisZubchenko** [co-IT/ngrx-ducks#30]
Usually, we use several actions for a request:
```
@StoreFacade
class UserFacade {

  loadRequest = createDuck('[User] Load Request');
  loadSuccess = createDuck('[User] Load Success', dispatch<User>());
  loadFailure = createDuck('[User] Load Failure');

}
```
How about allowing to group these actions?
```
@StoreFacade
class UserFacade {

  load = { // Or createDuckGroup...
    request: createDuck('[User] Load Request'),
    success: createDuck('[User] Load Success', dispatch<User>()),
    failure: createDuck('[User] Load Failure')
  }

}
```
It already works when dispatching whilst `getActions` result does not support that:
```
this.userFacade.load.request.dispatch(); // OK
userActions.load.request // Error
```

**Comment by GregOnNet** [comment]
Hi,

we recently discussed grouping of actions in #4.
I dropped the idea since the code would become less readable if `createDuck` is configured with a reducer-Function.

The good thing about grouping is, better naming possibilities.
The bad thing, at least in my view is, worse code readability.

Have you further arguments why grouping would make sense?

**Comment by BorisZubchenko** [comment]
@GregOnNet #4 looks complex to me.
My idea is to just add "syntax" sugar to the existing implementation.

> I dropped the idea since the code would become less readable if createDuck is configured with a reducer-Function.

If I understood correctly, I don't see any readability worsening since we can still use blank lines as before:
```
  load = createDuckGroup({

    request: createDuck('[User] Load Request'),

    success: createDuck(
        '[User] Load Success',
        (state: UserState, user: User) => ({ ...state, user })
    ),

    failure: createDuck('[User] Load Failure')

  });
```

**Comment by GregOnNet** [comment]
It becomes messy if the reducer is a bit more complex.
e.g.

```ts
  load = createDuckGroup({

    request: createDuck('[User] Load Request'),

    success: createDuck(
        '[User] Load Success',
        (state: UserState, user: User) => ({ 
           ...state,
           user,
           callState: CallState.LOADED
         })
    ),

    failure: createDuck('[User] Load Failure', 
        (state: UserState, user: User) => ({ 
           ...state,
           callState: CallState.ERROR
         })
    )
  });
```

**Comment by BorisZubchenko** [comment]
Agree, but without groups it is still messy:
```
loadRequest: createDuck('[User] Load Request'),

loadSuccess: createDuck(
    '[User] Load Success',
    (state: UserState, user: User) => ({ 
       ...state,
       user,
       callState: CallState.LOADED
     })
),

loadFailure: createDuck('[User] Load Failure', 
    (state: UserState, user: User) => ({ 
       ...state,
       callState: CallState.ERROR
     })
)
```
Tradeoffs:
* With my proposal there are extra indentations. 
* Current implementation forces us to add `load` prefix.

However, it seems more convenient to me since we can get tips by typing `load.`.

P.S. Forms, for example. We would rather encapsulate similar fields to a new group:
```
form = this.fb.group({
  firstName: '',
  middleName: '',
  lastName: '',
  // Other fields here
});

// Or

form = this.fb.group({
  name: this.fb.group({
    first: '',
    middle: '', // Can be complex too (many validators etc.)
    last: ''
  }),
  // Other fields here
})
```
(This is not a rule of thumb but matter of taste of course.)

**Comment by BorisZubchenko** [comment]
However, groups can be extracted to a constant (not sure 'bout typings):
```
@StoreFacade
export class UserFacade {

  load = loadActions;

}

const loadActions: DuckGroup = createDuckGroup({...});
```

**Comment by GregOnNet** [comment]
`getReducer` needs to be enhanced.

**Comment by GregOnNet** [comment]
reopen due to runtime error in example app

**Comment by GregOnNet** [comment]
Hi @BorisZubchenko,

grouping is now possible.
Have fun with it. :)

DEMO: https://stackblitz.com/edit/ngrx-ducks-9

**Comment by BorisZubchenko** [comment]
@GregOnNet, unfortunately, I [can't get an action group](https://stackblitz.com/edit/ngrx-ducks-9-4wz2nh?file=src/app/counter/store/counter/counter.facade.ts) (the last line) from `getActions` :(

**Comment by GregOnNet** [comment]
That's right.
It is a bug.

**Comment by GregOnNet** [comment]
Hi @BorisZubchenko,

sorry for the long delay.
I had a look at the bug again and did the implementation for `getActions`.

It has been released with NgRx Ducks 10.1.
An example can be found here: https://stackblitz.com/edit/ngrx-ducks-10?file=src/app/counter/store/counter/counter.effects.ts
